api/envoy/config/listener/v3/listener.proto

```diff
@@ -120,6 +122,79 @@ message Listener {
   // :ref:`FAQ entry <faq_how_to_setup_sni>`.
   repeated FilterChain filter_chains = 3;
 
+  // :ref:`Unified matcher <arch_overview_matching>` resolving the filter chain name from the network properties. This matcher is used as a replacement
+  // for the per-filter chain match condition
+  // :ref:`filter_chain_match <envoy_v3_api_field_config.listener.v3.FilterChain.filter_chain_match>`.
+  // If specified, all :ref:`filter_chains <envoy_v3_api_field_config.listener.v3.Listener.filter_chains>` must
+  // have non-empty and unique :ref:`name <envoy_v3_api_field_config.listener.v3.FilterChain.name>` fields and omit
+  // :ref:`filter_chain_match <envoy_v3_api_field_config.listener.v3.FilterChain.filter_chain_match>` field.
+  //
+  // The list of matching inputs is available :ref:`here <arch_overview_matching_network_inputs>`.
+  //
+  // Example: The following matcher selects three filter chains as follows:
+  //
+  // * if the destination port is 80, then the filter chain "http" is selected;
+  // * if the destination port is 443 and the source IP is in the range 192.0.0.0/2, then the filter chain "internal" is selected;
+  // * otherwise, if the destination port is 443, then the filter chain "https" is selected;
+  // * otherwise, the default filter chain is selected (or the connection is rejected without the default filter chain).
```

Comment:
We still have an `on_no_match` field in the matcher, do you think it would be better to omit `default_filter_chain` together with `filter_chain_match`?

Reply:
Yeah, the default filter chain is unnecessary with the unified matcher. We can outright reject that has both set. Would that be cleaner?

Reply:
Sure. I am just a little bit confused by `the default filter chain is selected`.

Reply:
Updated and simplified.